carbonmark/lib/getPoolData.ts

```diff
@@ -9,3 +12,29 @@ export const getPoolTokenType = (pool: Uppercase<PoolToken>): CarbonToken =>
 
 export const getDefaultPoolFromPrices = (prices: Price[]) =>
   prices.find((p) => p.isPoolDefault);
+
+export const getPoolApprovalValue = (
+  cost: string,
+  paymentMethod: CarbonmarkPaymentMethod,
+  maxDecimals?: number
```

Comment:
we should use `getTokenDecimals` and derive it from the payment method, not a configurable param

Reply:
My bad on this one, removed `maxDecimals` and now using `getTokenDecimals` from the payment method.